src/hooks/useOperatorOrder.ts

```diff
@@ -1,30 +1,72 @@
 import { useCallback, useEffect, useState } from 'react'
 
-import { Order, getOrder } from 'api/operator'
+import { Order, getOrder, RawOrder } from 'api/operator'
 
 import { transformOrder } from 'utils'
 
 import { useNetworkId } from 'state/network'
 
 import { useMultipleErc20 } from './useErc20'
+import { Network } from 'types'
+
+const NETWORK_ID_SEARCH_LIST = [Network.Mainnet, Network.xDAI, Network.Rinkeby]
 
 type UseOrderResult = {
   order: Order | null
   error?: string
   isLoading: boolean
+  errorOrderPresentInNetworkId: Network | null
   forceUpdate: () => void
 }
 
-export function useOrder(orderId: string, updateInterval = 0): UseOrderResult {
+interface GetOrderResult {
+  order: RawOrder | null
+  errorOrderPresentInNetworkId?: Network
+}
+
+async function _getOrder(
+  networkId: Network,
+  orderId: string,
+  networkIdSearchListRemaining: Network[],
+): Promise<GetOrderResult> {
+  const order = await getOrder({ networkId, orderId })
```

Comment:
I'm wondering, from the UX point of view, wouldn't it be better if we fetch all networks at once instead of trying sequentially?

Or only after the initial network id query fails.

Not sure, just thinking out loud.

Reply:
I saw it's pretty fast. So I just felt it was fine. I can happily change this if ends up being not performant or something. but is a good point. Being sequential also do less calls. I look in order of relevance (mainnet, xdai, rinkeby)